### DISCUSSION TITLE
PDF to TIFF conversion creates black boxes

### DISCUSSION THREAD

**Discussion by jrmccabe** [ImageMagick/ImageMagick#4760] [Help]
[1973-18-2.pdf](https://github.com/ImageMagick/ImageMagick/files/7943136/1973-18-2.pdf)
The PDF file attached comes out with black boxes for all the PDF layers.

I am using 
convert -density 300 1973-18-2.pdf 1973-18-2.tif

Any help would be appreciated!

**Comment by fmw42** [comment]
Works fine for me on IM 6.9.12-34 Mac OSX Sierra with Ghostscript 9.54 and libtiff 4.3.0. This is my command:

`convert -density 300 1973-18-2.pdf -background white -alpha background -alpha off x.tif`

[x.tif.zip](https://github.com/ImageMagick/ImageMagick/files/7943311/x.tif.zip)

**Reply by jrmccabe** [reply]
Thanks for the feedback!

I now realize how poorly I formatted the initial post :(

Trying the -background white -alpha background -alpha off I still get the black blocks.

Running on Windows 10, ImageMagick 7.1.0-20 Q16 x64 (Portable)
The machine is fairly locked down so I have few options for installing or updating software.

Here is a screen shot of my resullt - can't upload TIF file:
![image](https://user-images.githubusercontent.com/98476035/151206383-163413b5-8558-4973-9a9f-efaf8cfc4712.png)

**Reply by fmw42** [reply]
Do you have Ghostscript installed and if so, what version?

**Reply by fmw42** [reply]
What do you get from 

`magick -version`

**Reply by jrmccabe** [reply]
Version: ImageMagick 7.1.0-20 Q16 x64 2022-01-22 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC OpenCL 
Delegates (built-in): bzlib cairo freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193030709)

**Reply by jrmccabe** [reply]
> Do you have Ghostscript installed and if so, what version?

No, no gs installed.

**Reply by fmw42** [reply]
What version of Ghostscript and what version of libtiff?

**Reply by fmw42** [reply]
Your version information says gslib.  That is Ghostscript.

**Reply by jrmccabe** [reply]
I am using a Portable version of IM, so I do not have GhostScript installed and am unsure how to check on libtiff.  There is no reference to libtiff in the files provided by IM Portable.

**Reply by fmw42** [reply]
You may see a version number when you do 

`magick -list format`

Look for the line that says TIFF on the right side will be the version number. I am not sure if Ghostscript will show its version. But you might get it by

`gs --version`

or whatever is the equivalent on Windows.

**Reply by jrmccabe** [reply]
Ok - the -list format shows 

TIFF* as LIBTIFF, Version 4.3.0

**Reply by fmw42** [reply]
try

`gs --version`

**Reply by fmw42** [reply]
What is your platform?

**Reply by fmw42** [reply]
What is your IM version?

**Comment by snibgo** [comment]
> Here is a screen shot of my resullt - can't upload TIF file:

A screen shot, using what program to view the TIF file? The PDF page is mostly transparent, but your resulting PNG has no transparency.

**Reply by jrmccabe** [reply]
The screen shot is the TIF display, added ".txt" to the end of the tif file, not sure if you can download and remove the ".txt" to see the actual tif.
[1973_18_2.tif.txt](https://github.com/ImageMagick/ImageMagick/files/7944104/1973_18_2.tif.txt)

Can't upload TIF file type directly

**Reply by fmw42** [reply]
Zip the tiff file and then upload it.  Changing suffix is not a good idea.

**Reply by jrmccabe** [reply]
Here is the zipped version:
[1973_18_2.tif.zip](https://github.com/ImageMagick/ImageMagick/files/7944168/1973_18_2.tif.zip)

**Reply by fmw42** [reply]
You have not answered my questions about your

IM version and platform and version of Ghostscript!

I suspect one needs to be upgraded.

**Reply by jrmccabe** [reply]
I do not have GhostScript installed

Running on Windows 10, ImageMagick 7.1.0-20 Q16 x64 (Portable)
The machine is fairly locked down so I have few options for installing or updating software.

**Reply by fmw42** [reply]
What do you get from 

`magick -version`

If that shows gslib under Features, then you have Ghostscript. If you do not have Ghostscript, then that is the reason, since ImageMagick requires Ghostscript to read PDF files properly.

**Reply by jrmccabe** [reply]
Looks like Ghostscript is the dependency here.  

Magick -version only shows "Cipher DPC OpenCL" but no Ghostscript.

I will see if I can get GS installed and update this thread with progress.

A HUGE thank you for your time and expertise!

**Reply by fmw42** [reply]
Are you saying there is no list of delegates following "Delegates:".  It should say something like:

```
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype gslib gvc heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr png ps raw rsvg tiff webp x xml zlib
Compiler: gcc (4.2)
```

In the list, should be `gslib` or `gs`

**Reply by fmw42** [reply]
You need things like freetype, gslib, jpeg, lcms, png, tiff. If you do not have those, then something is wrong, unless you compiled from source. If you installed a binary, then it should have most of the ones that I listed above.

**Comment by jrmccabe** [comment]
Here’s what I get from -version


Version: ImageMagick 7.1.0-20 Q16 x64 2022-01-22 https://imagemagick.org <https://imagemagick.org/>
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php <https://imagemagick.org/script/license.php>
Features: Cipher DPC OpenCL
Delegates (built-in): bzlib cairo freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193030709)

**Reply by fmw42** [reply]
So it does list gslib.  So you have Ghostscript.  Now you need to try to find its version.  Does

`gs --version`

work to tell you the version?

**Reply by fmw42** [reply]
As a note, the delegates page on the IM web site at https://download.imagemagick.org/ImageMagick/download/delegates/ shows Ghostscript 9.22. That is rather old. Perhaps that is what you got from your install. You might want to ask the IM developers to install a more current version of Ghostscript in the binaries for Windows.  Perhaps they already have in the current version of IM? Your IM is a few releases old.

**Comment by jrmccabe** [comment]
There is no “gs” available on my Windows machine. :/

**Reply by fmw42** [reply]
Sorry, I do not know or use Windows. But there should be a way to get the version.  Perhaps one of the Windows user's can help in this regard and show you how to find the version of Ghostscript on Windows.

**Comment by fmw42** [comment]
Try

`gswin64c -version`

or

`gswin32c -version`

**Comment by jrmccabe** [comment]
Not recognized on either command.

**Comment by snibgo** [comment]
The commands "gs -version" or "gswin64c -version" etc would be useful if Ghostscript was installed as an external delegate, an executable program. Also, "magick -verbose in.pdf NULL:" would give informaton about Ghostscript as an external delegate.

@jrmccabe reports that "-version" says Ghostscript is installed as an _internal_ delegate, "gslib". This means the GS library is linked to IM, so there is no need for an external GS program.

IM doesn't seem to report the version of gslib.

**Reply by fmw42** [reply]
Too bad.  I suspect IM has installed an older version of Ghostscript, since the list of delegates on the delegates page says 9.22 and currently Ghostscript is at 9.55 (see https://www.ghostscript.com/releases/gsdnld.html)

**Comment by jrmccabe** [comment]
Well, I installed IM on OSX and can convert the images without issue.  Having to work with corporate infrastructure is as much fun as a day at the DMV >:(

Thanks all for the help!
